compiler/rustc_transmute/src/lib.rs

```diff
@@ -148,7 +149,17 @@ mod rustc {
             );
 
             let variant = adt_def.non_enum_variant();
-            let fields = c.to_valtree().unwrap_branch();
+            let fields = match c.try_to_valtree() {
+                Ok(ValTree::Branch(branch)) => branch,
+                _ => {
```

Comment:
random comment: since you're not using the allocated error message, maybe `try_to_valtree` could return an Option instead.